functions/src/models/requests/timeline.ts

```diff
@@ -1,19 +1,35 @@
 import { FirestoreDataConverter } from '@google-cloud/firestore';
-import { IsObject } from 'class-validator';
+import { IsEnum, IsNotEmptyObject, IsObject, IsOptional, ValidateNested } from 'class-validator';
 import { firestore } from 'firebase';
 
 import { IOffer, Offer } from '../offers';
 import { IRequest, Request } from './index';
+import { IUser, User } from '../users';
 import Timestamp = firestore.Timestamp;
 import DocumentData = firestore.DocumentData;
 import DocumentReference = firestore.DocumentReference;
 import QueryDocumentSnapshot = firestore.QueryDocumentSnapshot;
```

Comment:
I think we must import these types from '@google-cloud/firestore'. 

I got a silly issue while I was trying to add another operation to create a timeline at the request create trigger. If we try to get ref from a snapshot, i.e 'snapshot.ref', it's a FirebaseFirestore.DocumentReference not a firebase.firestore.DocumentReference. So, it's mismatching and causes type errors.

Did I miss something? @sabind @ashwinkjoseph

Reply:
yea so the models use Firebase and the functions use Firebase Admin so yes we have to make sure we import from the @google-cloud/firestore to be consistent.

Reply:
Also, we have to change the type of operations array. `FirebaseFirestore.DocumentReference` returns `Promise<WriteResult>` not `Promise<void>`

Reply:
For that one we simply do a `.then(() => return Promise.resolve())` to throw away the `WriteResult`